src/libraries/System.Runtime.InteropServices/tests/ComDisabled/System/Runtime/InteropServices/Marshal/GetIDispatchForObjectTests.Windows.cs

```diff
@@ -0,0 +1,18 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using System.Runtime.InteropServices.Tests.Common;
+using Xunit;
+
+namespace System.Runtime.InteropServices.Tests
+{
+    public partial class GetIDispatchForObjectTests
+    {
+        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsNotWindowsNanoServer))]
+        public void GetIDispatchForObject_DispatchObject_Fail()
+        {
+            Assert.Throws<NotSupportedException>(() => Marshal.GetIDispatchForObject(new object()));
```

Comment:
Do we care about checking the message in the exception (as indication that it is specifically for COM being disabled)?

The binary formatter tests do something like that by checking for the expected aka.ms link:
https://github.com/dotnet/runtime/blob/916d58a0ba785db4aead6974bdfaa6157fb81614/src/libraries/System.Runtime.Serialization.Formatters/tests/DisableBitTests.cs#L50-L51